services/spar/test-integration/Util/SCIM.hs

```diff
@@ -284,18 +333,46 @@ acceptScim = accept "application/scim+json"
 scimUserId :: SCIM.StoredUser -> UserId
 scimUserId storedUser = either err id (readEither id_)
   where
-    id_ = unpack (SCIM.id (SCIM.thing storedUser))
+    id_ = cs (SCIM.id (SCIM.thing storedUser))
     err e = error $ "scimUserId: couldn't parse ID " ++ id_ ++ ": " ++ e
 
--- | Check that some properties match between an SCIM user and a Brig user.
-userShouldMatch
-    :: (HasCallStack, MonadIO m)
-    => SCIM.StoredUser -> User -> m ()
-userShouldMatch scimStoredUser brigUser = liftIO $ do
+-- | There are a number of user types that all partially map on each other.  This class provides a
+-- uniform interface to express those mappings.
+class UserShouldMatch u1 u2 where
+  userShouldMatch
+    :: forall m. (HasCallStack, MonadIO m)
+    => u1 -> u2 -> m ()
```

Comment:
Maybe replace it with `u1 -> u2 -> Assertion`? It means you will have to use `liftIO` in more places, but on the other hand it's more understandable.